jax/_src/numpy/ufuncs.py

```diff
@@ -218,6 +218,17 @@ def _arccosh(x: ArrayLike, /) -> Array:
 atan2 = _one_to_one_binop(getattr(np, "atan2", np.arctan2), lax.atan2, True)
 
 
+def print_binary(x):
+  if np.isscalar(x):
+    return bin(int(x))
+
+  elif x.ndim == 1:
+    return [bin(int(val)) for val in x]
+
+  else:
+    return [print_binary(sub_arr) for sub_arr in x]
```

Comment:
I don't think this should be a public API. We should define it as a 2-line helper function in the docstring.

Reply:
Thanks for your help! Would you happen to know the proper syntax for this?

If I try:
```py
>>>  def print_binary(x):
>>>    return [bin(int(val)) for val in x]
```
it renders well but I get this in the CI:
```
279     >>> def print_binary(x):
UNEXPECTED EXCEPTION: IndentationError('expected an indented block', ('<doctest jax._src.numpy.ufuncs.right_shift[0]>', 1, 21, 'def print_binary(x):\n'))
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/doctest.py", line 1334, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest jax._src.numpy.ufuncs.right_shift[0]>", line 1
    def print_binary(x):
                        ^
```
```py
>>> def print_binary(x):
  return [bin(int(val)) for val in x]
```
It seems to render poorly.